Add an offboarding checklist (#390)
kyma-project/test-infra: docs/prow/offboarding-checklist.md

# Offboarding Checklist

> When someone having access to the Prow cluster leaves the project, we need to apply the following steps to keep our assets secured.

Reviewer: ```suggestion
When someone with access to the Prow cluster leaves the project, we must apply the following steps to keep our assets secured.
```

> ## Remove the person from Google project

Reviewer: ```suggestion
## Remove the person from the Google project
```

> The person should be removed from the Google project `kyma-prow` immediately. Follow [this document](https://cloud.google.com/iam/docs/granting-changing-revoking-access) to revoke access.

Reviewer: ```suggestion
Remove the person from the `kyma-prow` Google project immediately. Follow [this document](https://cloud.google.com/iam/docs/granting-changing-revoking-access) to revoke access.
```

> ## Rotate all the secrets

Reviewer: ```suggestion
## Rotate all the Secrets
```

> All the secrets that were valid when the person was in the project should be rotated. Follow [Prow secret management](./prow-secrets-management.md) to create a new key ring and new secrets. Then, use [secrets populator](./../../development/tools/cmd/secretspopulator/README.md) to update all the secrets on Prow cluster.

Reviewer: ```suggestion
All the Secrets that were valid when the person was in the project must be rotated. Follow the [Prow secret management](./prow-secrets-management.md) to create a new keyring and new Secrets. Then, use [Secrets populator](./../../development/tools/cmd/secretspopulator/README.md) to update all the Secrets in the Prow cluster.
```
